starlette/templating.py

```diff
@@ -65,7 +65,12 @@ def _create_env(
         self, directory: typing.Union[str, PathLike], **env_options: typing.Any
     ) -> "jinja2.Environment":
         @pass_context
-        def url_for(context: dict, name: str, **path_params: typing.Any) -> str:
+        def url_for(context: dict, *args: str, **path_params: typing.Any) -> str:
+            if len(args) < 1:
+                raise TypeError("Missing route name as the second argument.")
+            elif len(args) > 1:
+                raise TypeError("Invalid positional argument passed.")
+            name = args[0]
             request = context["request"]
             return request.url_for(name, **path_params)
```

Comment:
Can you explain the change? Won't this work?
```suggestion
        def url_for(context: dict, *args: str, **path_params: typing.Any) -> str:
            request = context["request"]
            return request.url_for(*args, **path_params)
```

Reply:
It'd work. But we couldn't be sure that the caller of the function uses it as intended.
In the original signature it's obvious, that the first argument must the the routes name, as its the name of the argument. But by removing the variable name (replacing  with `*args`) the caller now has no indication what to do.
Those `TypeErrors` are our *communication* to future starlette users.

Reply:
Yeah but you can (or already did) the validation in `request.url_for`, right? Isn't this duplicated?

Reply:
Yes, you are right. The check can be removed and the one in `request.url_for()` will catch it.
But I fear, that it'll be *confusing* to get the exception message `"url_for() takes exactly one positional argument"` from `request.url_for()` when using the `templating....url_for()` with wrong arguments, as *that* requires exactly 2 positional arguments.
I am unsure what is the best way forward: reduced code vs explicit error messages. As in the long run the maintainers will have to deal with this decision, I'll defer it to them. If you are one of them (or simply confident enough), I'll do the change, no more discussion, just write "do it" :)

Reply:
@aminalaee ping. Please decide. I'll follow your decision.

Reply:
I think I'd use the `url_for` validation and don't do it in the templating. I was waiting for other feedback.

Reply:
Done (afefc16).